### PR TITLE
[15.0][ADD] stock_request_bom: Integrate BOM auto-completion in stock requests

### PR DESCRIPTION
**Pull request by JoanSForgeFlow**
This module adds BOM auto-completion functionality to stock request orders. 

It introduces `Product BOM` and `Quantity BOM` fields, allowing automatic population of stock request lines based on BOM components and the specified quantity.

### CODE REVIEW

File: stock_request_bom/models/stock_request_order.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            return
+
+        bom = (
+            self.env["mrp.bom"]._bom_find(self.product_bom_id).get(self.product_bom_id)
+        )
+        if not bom:
+            raise UserError(_("No BOM found for the selected product."))
+        # Clear the existing stock request lines
+        self.stock_request_ids = [(5, 0, 0)]
+        bom_lines = self._get_bom_lines_recursive(bom, self.quantity_bom)
```
Comment (JordiMForgeFlow) [inline stock_request_bom/models/stock_request_order.py:79]: you could try using the standard explode method

Comment (JoanSForgeFlow) [reply stock_request_bom/models/stock_request_order.py]: Done! Thank you :+1:

Comment (JoanSForgeFlow) [reply stock_request_bom/models/stock_request_order.py:79]: Done! Thank you

**Review by JordiMForgeFlow — APPROVED** [review]
code review

**Review by AaronHForgeFlow — APPROVED** [review]
:+1:  functional test

**Review by LoisRForgeFlow — APPROVED** [review]
Functional review :+1:

**Pull request merged**